logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/handler/ButtonActionConfirmationhandler.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.sync.receiver.handler
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.CompositeMessageRepository
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.functional.Either
+
+internal interface ButtonActionConfirmationHandler {
+    suspend fun handle(
+        conversationId: ConversationId,
+        messageContent: MessageContent.ButtonActionConfirmation
+    ): Either<StorageFailure, Unit>
+}
+
+internal class ButtonActionConfirmationHandlerImpl internal constructor(
+    private val compositeMessageRepository: CompositeMessageRepository
+) : ButtonActionConfirmationHandler {
+
+    override suspend fun handle(
```

Comment:
Don't we need to verify that the message came from the self user or ignore otherwise?

Reply:
Good catch, we should verify that the sender is the same sender of the original composite message, not self user

Reply:
I'm confused. For example, isn't this a message you send when you make a selection on a poll. Doesn't everyone in the group potentially make a selection, not only the original sender of the composite message?

Reply:
> Doesn't only a service care about the answer? But as a client we want to sync the button selection state across clients.

the service will send the confirmation message to all of our clients so there is no need to do any extra sync 

the flow works like this:
user click a button -> app will send a `ButtonAction` message with the button id to the service only -> the service will handle the action and send the `ButtonActionConfirmation` message to all of our clients

Reply:
Thanks for explaining! All makes sense now